api/swagger-spec/oapi-v1.json

```diff
@@ -13906,6 +13906,10 @@
       "type": "string",
       "description": "type of source control management system"
      },
+     "dockerfile": {
+      "type": "string",
+      "description": "the contents of a Dockerfile to build; FROM and ENV may be overriden if you have specified 'from' and 'env' on the build strategy"
```

Comment:
@mfojtik is right. We may replace the FROM, and we insert two ENV instructions: one right below FROM, with the contents from buildConfig.dockerStrategy.env, and one at the very bottom of the Dockerfile with OpenShift/build-related information (as well as labels).